service/core/action/medium/list.go

```diff
@@ -39,14 +42,45 @@ func list(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	searchQuery := r.URL.Query().Get("q")
+	var filteredMediumIDs []uint
+
+	if searchQuery != "" {
+		filters := fmt.Sprint("space_id=", sID)
+
+		var hits []interface{}
+
+		hits, err = meili.SearchWithQuery(searchQuery, filters, "medium")
+		if err != nil {
+			loggerx.Error(err)
+			errorx.Render(w, errorx.Parser(errorx.InternalServerError()))
+			return
+		}
+
+		filteredMediumIDs = meili.GetIDArray(hits)
+		if len(filteredMediumIDs) == 0 {
+			renderx.JSON(w, http.StatusOK, paging{
+				Nodes: make([]model.Medium, 0),
+				Total: 0,
+			})
+			return
+		}
+	}
+
 	result := paging{}
 	result.Nodes = make([]model.Medium, 0)
 
 	offset, limit := paginationx.Parse(r.URL.Query())
 
-	err = config.DB.Model(&model.Medium{}).Where(&model.Medium{
+	tx := config.DB.Model(&model.Medium{}).Where(&model.Medium{
 		SpaceID: uint(sID),
-	}).Count(&result.Total).Order("id desc").Offset(offset).Limit(limit).Find(&result.Nodes).Error
+	}).Order("id desc")
+
+	if len(filteredMediumIDs) > 0 {
+		err = tx.Where(filteredMediumIDs).Count(&result.Total).Offset(offset).Limit(limit).Find(&result.Nodes).Error
+	} else {
+		err = tx.Count(&result.Total).Offset(offset).Limit(limit).Find(&result.Nodes).Error
+	}
```

Comment:
Can we remove the else condition and do something like this
````. 
if len(filteredMediumIDs) > 0 {
		err = tx.Where(filteredMediumIDs)
}
err = tx.Count(&result.Total).Offset(offset).Limit(limit).Find(&result.Nodes).Error````

Reply:
I tried it but not working as expected